packages/pass-style/src/remotable.js

```diff
@@ -167,15 +166,26 @@ const checkRemotable = (val, check) => {
   return result;
 };
 
-/** @type {MarshalGetInterfaceOf} */
+/**
+ * Simple semantics, just tell what interface (or undefined) a remotable has.
+ * @type {{
+ * <T extends string>(val: import('./types.js').TaggedRecord<any, T>): T;
+ * (val: any): any;
+ * }}
```

Comment:
Cool! Can we narrow further?
```suggestion
/**
 * Simple semantics, just tell what interface (or undefined) a remotable has.
 * @type {{
 * <T extends string>(val: import('./types.js').TaggedRecord<any, T>): T;
 * (val: any): string | undefined;
 * }}
```

Reply:
Since remotable tags are only for debugging, I don't think we should use them for static typing.

Reply:
> Can we narrow further?

I had tried that and it caused some problems. I'll document.

> Since remotable tags are only for debugging, I don't think we should use them for static typing.

That's what this doc says,
https://github.com/endojs/endo/blob/40f6cf8794f9c67b3f5a65f01b7037ab960149bc/packages/patterns/src/types.js#L310-L313

but when we changed the label it failed rehydration,
https://github.com/Agoric/agoric-sdk/blob/62a7c9c42007020994f6036cf4090590e4811b8e/packages/zoe/src/zoeService/startInstance.js#L65

Is that a bug?

Reply:
> when we changed the label it failed rehydration, https://github.com/Agoric/agoric-sdk/blob/62a7c9c42007020994f6036cf4090590e4811b8e/packages/zoe/src/zoeService/startInstance.js#L65
> 
> Is that a bug?

Plausibly, although not related to this PR. The fundamental issue is that [agoric-sdk doesn't allow a new vat incarnation to change `stateShape`](https://github.com/Agoric/agoric-sdk/issues/7337#issuecomment-1514195810), and adding/removing/altering the label of a Matcher\<'remotable'> _is_ a change (albeit a benign one from internal structure like `{ [Symbol.for("passStyle")]: "tagged", [Symbol.toStringTag]: "match:kind", payload: "remotable" }` to `{ [Symbol.for("passStyle")]: "tagged", [Symbol.toStringTag]: "match:remotable", payload: { label: "$label" } }`). This will need to be allowed in the fix for https://github.com/Agoric/agoric-sdk/issues/7337 / https://github.com/Agoric/agoric-sdk/issues/7407 .